### DISCUSSION TITLE
Slow consult-man

### DISCUSSION THREAD

**Discussion by nicolasdespres** [minad/consult#1133] [Q&A] [answered]
`consult-man` is slow on my machine. `man` is slow too but only the first time. The culprit is `man -k` which is slow on my machine (about 2s). `man` cache the result of `man -k` so that it is slow only when I start it but not every time if I change the search string. If I comment the line `'Man-completion-cache   ; Don't cache across calls` in `man` function definition, I get a slow `man` only at the first invocation, but `consult-man` keeps being slow when I change the search string from `ls` to `printf` for instance (I have set `(setq consult-async-min-input 2)`). It seems that `consult-man` runs `man -k` each time the search string changes.

**Comment by minad** [comment]
This is somewhat by design. `consult-man` assumes that `man -k` is fast enough for an asynchronous search. If it is not, the command is not supported. Which man package do you use? As far as I know `man -k` uses an index internally, at least on some systems. For some input `man -k printf` it returns pretty quickly on my system.

**Comment by nicolasdespres** [comment, marked as answer]
Hi, thanks for replying so quickly. I used to use the default `man` command installed with macOS. I never really used `-k` and Zsh was kind of fast. I have now installed `man-db` which does provide an index. Now, it is fast. Sorry for the noise. consult is great. I love it :)

**Reply by hmelman** [reply]
Just to be explicit, on macOS you installed man-db and at least with brew's defaults it installs gnu man as `gman`.  Then you want to set `consult-man-args` to `"gman -k"` and probably `manual-program` to `"gman"`.

**Reply by nicolasdespres** [reply]
Yes, that's what I did.

```elisp
;; Use an indexed man db on OSX otherwise man -k is slow.
(when (executable-find "gman")
  (setq consult-man-args "gman -k")
  (setq manual-program "gman"))
```

I also ran `gmandb` to build the index.
